internal/messaging/response_handler.go

```diff
@@ -1,147 +1,536 @@
 
-	"github.com/ava-labs/avalanchego/ids"
-	"github.com/ava-labs/hypersdk/chain"
-	"github.com/ava-labs/hypersdk/codec"
-	"github.com/chain4travel/camino-messenger-bot/internal/tvm"
-	"github.com/chain4travel/caminotravelvm/actions"
-	"github.com/chain4travel/caminotravelvm/consts"
+	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	config "github.com/chain4travel/camino-messenger-bot/config"
+	"github.com/chain4travel/camino-messenger-bot/internal/evm"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
 	"go.uber.org/zap"
 )
 
-var _ ResponseHandler = (*TvmResponseHandler)(nil)
+var _ ResponseHandler = (*EvmResponseHandler)(nil)
 
 type ResponseHandler interface {
 	HandleResponse(ctx context.Context, msgType MessageType, request *RequestContent, response *ResponseContent)
+	HandleRequest(ctx context.Context, msgType MessageType, request *RequestContent) error
 }
-type TvmResponseHandler struct {
-	tvmClient *tvm.Client
+type EvmResponseHandler struct {
+	ethClient *ethclient.Client
 	logger    *zap.SugaredLogger
+	pk        *secp256k1.PrivateKey
+	cfg       *config.EvmConfig
 }
 
-func (h *TvmResponseHandler) HandleResponse(ctx context.Context, msgType MessageType, request *RequestContent, response *ResponseContent) {
+func (h *EvmResponseHandler) HandleResponse(ctx context.Context, msgType MessageType, request *RequestContent, response *ResponseContent) {
 	switch msgType {
 	case MintRequest: // distributor will post-process a mint request to buy the returned NFT
 		if h.handleMintRequest(ctx, response) {
 			return
 		}
-	case MintResponse: // provider will act upon receiving a mint response by minting an NFT
+	case MintResponse: // supplier will act upon receiving a mint response by minting an NFT
 		if h.handleMintResponse(ctx, response, request) {
 			return
 		}
 	}
 }
 
-func (h *TvmResponseHandler) handleMintResponse(ctx context.Context, response *ResponseContent, request *RequestContent) bool {
-	owner := h.tvmClient.Address()
+func (h *EvmResponseHandler) HandleRequest(_ context.Context, msgType MessageType, request *RequestContent) error {
+	switch msgType { //nolint:gocritic
+	case MintRequest:
+		request.BuyerAddress = crypto.PubkeyToAddress(h.pk.ToECDSA().PublicKey).Hex()
+	}
+	return nil
+}
+
+func (h *EvmResponseHandler) handleMintResponse(_ context.Context, response *ResponseContent, request *RequestContent) bool {
 	if response.MintResponse.Header == nil {
 		response.MintResponse.Header = &typesv1alpha.ResponseHeader{}
 	}
-	buyer, err := codec.ParseAddressBech32(consts.HRP, request.MintRequest.BuyerAddress)
+	bookingTokenABIFile := h.cfg.BookingTokenABIFile
+	abi, err := loadABI(bookingTokenABIFile)
```

Comment:
```suggestion
	abi, err := loadABI(h.cfg.BookingTokenABIFile)
```
or is there reason for not using cfg?